saleor/plugins/base_plugin.py

```diff
@@ -1098,12 +1099,12 @@ def save_plugin_configuration(
 
         cls.validate_plugin_configuration(plugin_configuration)
         cls.pre_save_plugin_configuration(plugin_configuration)
+        plugin_configuration.save()
```

Comment:
Why you moved it here? I moved it to line 1106 as `_append_config_structure` also updating the config structure, and if the `save` method is called here, the changes made in `_append_config_structure` are not saved.

Reply:
The config_structure is not a data that we should store in DB. It is a static data that is placed in `Plugin.CONFIG_STRUCTRUE`. 
IMHO `_append_config_structure` is a something that shouldn't be call here at all, as a main task of this method is attaching the config structure required for response send via API.  I've checked the code and I don't see any dependencies that would relay on the response from this method. We can drop it, but not in this PR.